### DISCUSSION TITLE
Mark a field as deprecated

### DISCUSSION THREAD

**Discussion by ryanleichty** [craftcms/cms#15632] [Ideas] [answered]
This is another idea coming from Sanity where you can [mark a field as deprecated](https://www.sanity.io/docs/schema-and-content-migrations#427cd19d541b).

- Mark the field as `deprecated`
- Add a `reason` for the deprecation (“Use the body field instead”).
- Optionally make the field read only

![image](https://github.com/user-attachments/assets/30b1ac1a-ad62-44eb-b240-f420c916d82c)

Quick mockup using existing elements:

![CleanShot 2024-08-30 at 13 05 39 2@2x](https://github.com/user-attachments/assets/bffa6f6c-61fb-4eb6-97ed-bce5f8ff010a)
![CleanShot 2024-08-30 at 13 13 48@2x](https://github.com/user-attachments/assets/26b60c9b-1338-48dd-b499-2a0588698b09)

**Comment by mmikkel** [comment]
I like this idea, but I think the ability to add _warnings_ and _tips_ to custom fields would cover the use case (aside from the read-only thing), and would probably be more useful vs. a dedicated feature to "deprecate" fields. I suggested this a while back: https://github.com/craftcms/cms/discussions/12821

That said, if it was possible to make custom fields read-only in field layouts (perhaps even conditionally), that would also be really cool. (FWIW, the [Incognito Field](https://plugins.craftcms.com/incognito-field?craft5) plugin can currently be used to create text fields that are conditionally read-only, hidden or disabled.)

**Reply by ryanleichty** [reply]
Yeah, I feel like it would for sure have to be within a field layout at minimum. Perhaps the warning or tip is included in a tooltip on top of a badge:

![CleanShot 2024-08-30 at 13 41 35@2x](https://github.com/user-attachments/assets/a9e24429-f66f-4c00-bcaa-72fe3b173d0a)

**Comment by brandonkelly** [comment, marked as answer]
Just added the ability to set tips and warnings on fields within field layout designers, for Craft 5.4 (1d3658d00b97c685fb192796695703b111e5fc85).

I think this makes more sense than an explicit “deprecated” field concept for a couple reasons:

- More versatile (there are a number of reasons you may want to warn authors about a particular field).
- “Deprecated” is developer lingo. Kindof doubt most authors would understand what it means, and there’s no good reason to start teaching them, when a more verbose explanation is likely needed anyway.

Here’s an example, combined with a condition that hides the field entirely when it doesn’t have a value:

![A field layout designer, with the settings slideout open for a “Subheading” field. The slideout includes a Warning value of “This field is no longer used.”, and has a condition that only shows the field when it already has a value.](https://github.com/user-attachments/assets/65bb2ce2-f0bd-4fcb-88fa-7955a32f953e)

**Reply by mmikkel** [reply]
Hooray! 👏 👏

**Reply by ryanleichty** [reply]
That was fast! 🚀

**Reply by brandonkelly** [reply]
Craft 5.4.0 is out with those new settings.
